skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/routes/SakBrev.kt

```diff
@@ -0,0 +1,138 @@
+package no.nav.pensjon.brev.skribenten.routes
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.ktor.server.util.*
+import no.nav.pensjon.brev.api.model.maler.BrevbakerBrevdata
+import no.nav.pensjon.brev.api.model.maler.Brevkode
+import no.nav.pensjon.brev.api.model.maler.RedigerbarBrevdata
+import no.nav.pensjon.brev.skribenten.auth.AuthorizeAnsattSakTilgang
+import no.nav.pensjon.brev.skribenten.db.Brevredigering
+import no.nav.pensjon.brev.skribenten.letter.Edit
+import no.nav.pensjon.brev.skribenten.services.BrevredigeringService
+import no.nav.pensjon.brev.skribenten.services.PenService
+import java.time.LocalDateTime
+
+fun Route.sakBrev(brevredigeringService: BrevredigeringService) =
+    route("/brev") {
+        post<OpprettBrevRequest> { request ->
+            val sak: PenService.SakSelection = call.attributes[AuthorizeAnsattSakTilgang.sakKey]
+
+            brevredigeringService.opprettBrev(call, sak, request.brevkode, request.saksbehandlerValg) {
+                OppprettBrevResponse(
+                    id = id.value,
+                    redigertBrev = redigertBrev,
+                    sistredigert = sistredigert,
+                    brevkode = Brevkode.Redigerbar.valueOf(brevkode),
+                    saksbehandlerValg = saksbehandlerValg,
+                )
+            }.map { brev ->
+                call.respond(HttpStatusCode.OK, brev)
+            }.catch { message, statusCode ->
+                call.application.log.error("$statusCode - Feil ved oppretting av brev ${request.brevkode}: $message")
+                call.respond(HttpStatusCode.InternalServerError, "Feil ved oppretting av brev.")
+            }
+        }
+
+        post<OppdaterBrevRequest>("/{brevId}") { request ->
+            val brevId = call.parameters.getOrFail<Long>("brevId")
+            brevredigeringService.oppdaterBrev(call, brevId, request.brevkode, request.saksbehandlerValg, request.redigertBrev) {
+                OppprettBrevResponse(
+                    id = brevId,
+                    redigertBrev = redigertBrev,
+                    sistredigert = sistredigert,
+                    brevkode = request.brevkode,
+                    saksbehandlerValg = saksbehandlerValg
+                )
+            }.map { brev ->
+                call.respond(HttpStatusCode.OK, brev!!)
```

Comment:
Bør ikke ha en not-null-assertion her. Svar heller med 404 om `brev` er null, siden det betyr at det ikke finnes i databasen (?).